### PR TITLE
MSEARCH-794: Initiate Full Reindex

### PR DESCRIPTION
**Pull request by mukhiddin-yusuf**
### Purpose
_Initiate and start new Full Reindex_

### Approach

1. Truncate tables: instance, item, holding, ranges (merge and upload), status
2. Request counts of records from mod-inventory-storage
3. Construct records' ID ranges and fill Merge Range table
4. Call mod-inventory-storage's POST /inventory-reindex-records/publish for constructed records' ranges

For ECS:

- repeat the steps 2-3 for each member tenant
- Do not allow running full reindex from member tenants

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [x] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [ ] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MSEARCH-794](https://folio-org.atlassian.net/browse/MSEARCH-794)

### Manual testing results:
tenant: diku
instance records count: 41
item records count: 25
holding records count: 21
REINDEX_MERGE_RANGE_SIZE: 10

image

image

image

ECS:
tanant: consortium (central)
instances count: 39
items count: 25
holdings count: 21

tenant: college
instances count: 2
items count: 0
holdings count: 1

tenant: university
instances count: 1
items count: 0
holdings count: 1

image

image

image

### CODE REVIEW

File: src/main/resources/changelog/changes/v4.0/update-reindex-status-trigger-fix.sql
Comment (mukhiddin-yusuf) [inline src/main/resources/changelog/changes/v4.0/update-reindex-status-trigger-fix.sql:1]: will remove this trigger and update the original ```update-reindex-status-trigger.sql``` with changes, I added this for testing purposes in the dev env

File: src/test/java/org/folio/search/service/reindex/jdbc/ReindexStatusRepositoryIT.java [hunk 1]
```diff
@@ -99,9 +99,10 @@ void setReindexUploadFailed() {
 
     // assert
     var statuses = repository.getReindexStatuses();
+
     assertThat(statuses)
       .hasSize(4)
-      .filteredOn(reindexStatus -> SUBJECT.equals(reindexStatus.getEntityType()))
+      .filteredOn(reindexStatus -> CLASSIFICATION.equals(reindexStatus.getEntityType()))
```
Comment (mukhiddin-yusuf) [inline src/test/java/org/folio/search/service/reindex/jdbc/ReindexStatusRepositoryIT.java:105]: revert to ```SUBJECT```

Comment (mukhiddin-yusuf) [inline src/test/java/org/folio/search/service/reindex/jdbc/ReindexStatusRepositoryIT.java:105]: revert to ```SUBJECT```

File: src/main/java/org/folio/search/configuration/AsyncConfig.java [hunk 2]
```diff
@@ -27,5 +28,10 @@ public Executor streamIdsExecutor() {
     executor.initialize();
     return executor;
   }
+
+  @Bean("reindexExecutor")
+  public FolioExecutor remappingExecutor() {
+    return new FolioExecutor(0, 1);
```
[the remark this answers is not in the document]
Comment (mukhiddin-yusuf) [reply src/main/java/org/folio/search/configuration/AsyncConfig.java:34]: does it really make sense to run another ```full reindex``` when one is running (as we truncate the tables and fill ranges for each separate run) ?

**Review by psmagin — APPROVED** [review]

**Pull request merged**